index out of range [0] with length 0 on linux arm64

Please answer the following before submitting your issue:

Note: Please include any substantial examples (debug session output,
stacktraces, etc) as linked gists.

1. What version of Delve are you using (`dlv version`)?
Delve Debugger
Version: 1.5.0
Build: af250d45126070adbc702fe4ae1b74f19dcfdf97

2. What version of Go are you using? (`go version`)?
tip
3. What operating system and processor architecture are you using?
linux arm64
4. What did you do?
debug regCase.go with the following command:
**GODEBUG=asyncpreemptoff=1 dlv exec ~/go-master/pkg/tool/linux_amd64/compile -- regCase.go**
regCase.go is as follow:

```
package foo

func sliceSum(a [][]int) int {
        sum := 0
        for _, c := range a {
                for _, d := range c {
                        if d > 0 {
                                sum += d
                        }
                }
        }
        return sum
}
```

5. What did you expect to see?
No panic
6. What did you see instead?
```
Warning: debugging optimized function
Values returned:
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/go-delve/delve/service/api.(*Variable).recursiveKind(0x4000460800, 0xdf2a0, 0xc849a0)
        /home/erifan01/gopath/src/github.com/go-delve/delve/service/api/prettyprint.go:294 +0x68
github.com/go-delve/delve/service/api.(*Variable).shouldNewlineArray(0x4000501730, 0xdf601, 0xc849a0)
        /home/erifan01/gopath/src/github.com/go-delve/delve/service/api/prettyprint.go:267 +0x4c
github.com/go-delve/delve/service/api.(*Variable).writeSliceOrArrayTo(0x4000501730, 0x7f5da0, 0x40038974a0, 0x1, 0x4002e73a68, 0x3)
        /home/erifan01/gopath/src/github.com/go-delve/delve/service/api/prettyprint.go:327 +0x30
github.com/go-delve/delve/service/api.(*Variable).writeSliceTo(0x4000501730, 0x7f5da0, 0x40038974a0, 0x101, 0x4002e73a68, 0x3)
        /home/erifan01/gopath/src/github.com/go-delve/delve/service/api/prettyprint.go:160 +0x68
github.com/go-delve/delve/service/api.(*Variable).writeTo(0x4000501730, 0x7f5da0, 0x40038974a0, 0x10100, 0x4002e73a68, 0x3)
        /home/erifan01/gopath/src/github.com/go-delve/delve/service/api/prettyprint.go:51 +0xcd0
github.com/go-delve/delve/service/api.(*Variable).writeStructTo(0x4000453800, 0x7f5da0, 0x40038974a0, 0x101, 0x4002e739d8, 0x2)
        /home/erifan01/gopath/src/github.com/go-delve/delve/service/api/prettyprint.go:193 +0x1e8
github.com/go-delve/delve/service/api.(*Variable).writeTo(0x4000453800, 0x7f5da0, 0x40038974a0, 0x10100, 0x4002e739d8, 0x2)
        /home/erifan01/gopath/src/github.com/go-delve/delve/service/api/prettyprint.go:86 +0xe04
github.com/go-delve/delve/service/api.(*Variable).writeTo(0x4000516450, 0x7f5da0, 0x40038974a0, 0x10100, 0x4002e739d8, 0x2)
        /home/erifan01/gopath/src/github.com/go-delve/delve/service/api/prettyprint.go:65 +0xb0c
github.com/go-delve/delve/service/api.(*Variable).writeStructTo(0x4000452000, 0x7f5da0, 0x40038974a0, 0x1b0101, 0x6e988c, 0x1)
        /home/erifan01/gopath/src/github.com/go-delve/delve/service/api/prettyprint.go:193 +0x1e8
github.com/go-delve/delve/service/api.(*Variable).writeTo(0x4000452000, 0x7f5da0, 0x40038974a0, 0x10100, 0x6e988c, 0x1)
        /home/erifan01/gopath/src/github.com/go-delve/delve/service/api/prettyprint.go:86 +0xe04
github.com/go-delve/delve/service/api.(*Variable).writeTo(0x4003e68fe8, 0x7f5da0, 0x40038974a0, 0x10101, 0x6e988c, 0x1)
        /home/erifan01/gopath/src/github.com/go-delve/delve/service/api/prettyprint.go:65 +0xb0c
github.com/go-delve/delve/service/api.(*Variable).MultilineString(0x4003e68fe8, 0x6e988c, 0x1, 0x1, 0x1)
        /home/erifan01/gopath/src/github.com/go-delve/delve/service/api/prettyprint.go:30 +0x70
github.com/go-delve/delve/pkg/terminal.printReturnValues(0x4003813860)
        /home/erifan01/gopath/src/github.com/go-delve/delve/pkg/terminal/command.go:2167 +0x120
github.com/go-delve/delve/pkg/terminal.printcontextThread(0x4003682be0, 0x4003813860)
        /home/erifan01/gopath/src/github.com/go-delve/delve/pkg/terminal/command.go:2177 +0x868
github.com/go-delve/delve/pkg/terminal.printcontext(0x4003682be0, 0x400070e180)
        /home/erifan01/gopath/src/github.com/go-delve/delve/pkg/terminal/command.go:2147 +0x104
github.com/go-delve/delve/pkg/terminal.(*Commands).next(0x40036cbaa0, 0x4003682be0, 0x0, 0xffffffffffffffff, 0x0, 0x0, 0x0, 0x0, 0x0, 0x7175a2, ...)
        /home/erifan01/gopath/src/github.com/go-delve/delve/pkg/terminal/command.go:1237 +0x198
github.com/go-delve/delve/pkg/terminal.(*Commands).CallWithContext(0x40036cbaa0, 0x400374aca4, 0x1, 0x4003682be0, 0x0, 0xffffffffffffffff, 0x0, 0x0, 0x0, 0x0, ...)
        /home/erifan01/gopath/src/github.com/go-delve/delve/pkg/terminal/command.go:501 +0xfc
github.com/go-delve/delve/pkg/terminal.(*Commands).Call(...)
        /home/erifan01/gopath/src/github.com/go-delve/delve/pkg/terminal/command.go:507
github.com/go-delve/delve/pkg/terminal.(*Term).Run(0x4003682be0, 0x0, 0x0, 0x0)
        /home/erifan01/gopath/src/github.com/go-delve/delve/pkg/terminal/terminal.go:248 +0x44c
github.com/go-delve/delve/cmd/dlv/cmds.connect(0x6eb32f, 0x4, 0x80efa0, 0x4000490500, 0x40004b1f80, 0x0, 0x0)
        /home/erifan01/gopath/src/github.com/go-delve/delve/cmd/dlv/cmds/commands.go:683 +0x104
github.com/go-delve/delve/cmd/dlv/cmds.execute(0x0, 0x4000485170, 0x2, 0x3, 0x40004b1f80, 0x0, 0x0, 0x0, 0x4000485170, 0x2, ...)
        /home/erifan01/gopath/src/github.com/go-delve/delve/cmd/dlv/cmds/commands.go:799 +0x864
github.com/go-delve/delve/cmd/dlv/cmds.New.func4(0x40004a3200, 0x4000485170, 0x2, 0x3)
        /home/erifan01/gopath/src/github.com/go-delve/delve/cmd/dlv/cmds/commands.go:213 +0x80
github.com/spf13/cobra.(*Command).execute(0x40004a3200, 0x40004850e0, 0x3, 0x3, 0x40004a3200, 0x40004850e0)
        /home/erifan01/gopath/pkg/mod/github.com/spf13/cobra@v0.0.0-20170417170307-b6cb39589372/command.go:647 +0x184
github.com/spf13/cobra.(*Command).ExecuteC(0x40004a26c0, 0x40004a26c0, 0x6ecc63, 0x6)
        /home/erifan01/gopath/pkg/mod/github.com/spf13/cobra@v0.0.0-20170417170307-b6cb39589372/command.go:733 +0x20c
github.com/spf13/cobra.(*Command).Execute(...)
        /home/erifan01/gopath/pkg/mod/github.com/spf13/cobra@v0.0.0-20170417170307-b6cb39589372/command.go:692
main.main()
        /home/erifan01/gopath/src/github.com/go-delve/delve/cmd/dlv/main.go:24 +0xc8
erifan01@erifan01-03-arm-vm:~/branch$ GODEBUG=asyncpreemptoff=1 dlv exec ~/go-master/pkg/tool/linux_arm64/compile -- regCase.go
Type 'help' for list of commands.


```

--- a/service/api/prettyprint.go
+++ b/service/api/prettyprint.go
@@ -291,6 +291,9 @@ func (v *Variable) recursiveKind() (reflect.Kind, bool) {
 		kind = v.Kind
 		if kind == reflect.Ptr {
 			hasptr = true
+			if len(v.Children) == 0 {
+				return kind, hasptr
+			}
 			v = &(v.Children[0])
 		} else {
 			break

